### DISCUSSION TITLE
Question: Adding location per server to serve static file.

### DISCUSSION THREAD

**Discussion by orion110217** [nginx-proxy/nginx-proxy#2241] [Q&A]
I have an idea how to do it in [nginx](https://www.nginx.com/resources/admin-guide/serving-static-content/), just confused on how can i do it using docker-gen. 

I have tried using Per-VIRTUAL_HOST default configuration, but doing so replaces the default configuration like servername etc. and only creating the one i wrote.

**Comment by arturluizbr** [comment]
Location per server i could do, my other need is to use multiples locations to different containers.

**Comment by orion110217** [comment]
Mind sharing your configuration?

**Comment by arturluizbr** [comment]
I just extended this image with:

```
RUN sed -i "s|location \/ {|{{ \$host_location := or (first (groupByKeys \$containers \"Env.HOST_LOCATION\")) \"\" }} \
	location \/{{ \$host_location }} {\
		{{ if ne \$host_location  \"\" }} \
		rewrite /$host_location(.*) /$1  break; \
		{{ end }} \
	 |g" /app/nginx.tmpl
```

Then, you can just use a HOST_LOCATION env on target containers and enjoy. =D

**Comment by cloud-rocket** [comment]
@arturluizbr, can you please share how to use your extension to configure static content location?

Thax

**Comment by arturluizbr** [comment]
@cloud-rocket iguess i didn't read that static content part.

What I did was to use a custom locations instead or root, probably in future i may need few customizations, if so, i'll make a pull request here. =D
